stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketObservability.java

```diff
@@ -0,0 +1,387 @@
+            observerContext.setConnectorName(SERVER_CONNECTOR_WEBSOCKET);
+            try {
+                observerContext.addTag(TAG_CONNECTION_ID,
+                                       connectionInfo.getWebSocketConnection().getChannelId());
+            } catch (IllegalAccessException e) {
+                observerContext.addTag(TAG_CONNECTION_ID, WEBSOCKET_UNKNOWN);
+            }
+            setObserveService(observerContext, connectionInfo);
+            Map<String, String> tags = observerContext.getTags();
+            Set<Tag> allTags = new HashSet<>(tags.size());
+            Tags.tags(allTags, observerContext.getTags());
+
+            MetricRegistry metricRegistry = DefaultMetricRegistry.getInstance();
+            //Increment current connections metric
+            metricRegistry.gauge(new MetricId(SERVER_CONNECTOR_WEBSOCKET + "_" + METRIC_CONNECTIONS,
+                                              METRIC_CONNECTIONS_DESC, allTags)).increment();
+
+            //Log connection
+            try {
+                logger.info("WS new connection established. connectionID: {}, service: {}",
+                            connectionInfo.getWebSocketConnection().getChannelId(),
+                            tags.get(TAG_SERVICE));
+            } catch (IllegalAccessException e) {
+                logger.info("WS new connection established. connectionID: {}, service: {}",
+                            WEBSOCKET_UNKNOWN,
+                            tags.get(TAG_SERVICE));
+            }
+        }
+    }
+
+    /**
+     * This method observes messages pushed (sent).
+     *
+     * @param type type of message pushed (text, binary, control, close).
+     * @param result whether the push was successful or not.
+     * @param connectionInfo information regarding connection.
+     */
+    public static void observePush(String type, String result, WebSocketOpenConnectionInfo connectionInfo) {
+        if (ObserveUtils.isObservabilityEnabled()) {
+            ObserverContext observerContext = new ObserverContext();
+            observerContext.setConnectorName(SERVER_CONNECTOR_WEBSOCKET);
+            //Define type of message (text, binary, control, clsoe) and result (successful, failed)
+            observerContext.addTag(TAG_MESSAGE_TYPE, type);
+            observerContext.addTag(TAG_KEY_RESULT, result);
+            try {
+                observerContext.addTag(TAG_CONNECTION_ID, connectionInfo.getWebSocketConnection().getChannelId());
+            } catch (IllegalAccessException e) {
+                observerContext.addTag(TAG_CONNECTION_ID, WEBSOCKET_UNKNOWN);
+            }
+            setObserveService(observerContext, connectionInfo);
+            Map<String, String> tags = observerContext.getTags();
+            Set<Tag> allTags = new HashSet<>(tags.size());
+            Tags.tags(allTags, observerContext.getTags());
+
+            MetricRegistry metricRegistry = DefaultMetricRegistry.getInstance();
```

Comment:
Same here. Shall we remove duplicates by creating new functions? Let's fix in all functions in this class